home/api/ApolloClient.ts

```diff
@@ -1,35 +1,46 @@
-import ApolloClient from 'apollo-boost';
-import { InMemoryCache, IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
+import { ApolloClient, InMemoryCache } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context';
+import { HttpLink } from '@apollo/client/link/http';
 
 import Store from '../redux/Store';
 import Config from './Config';
 import Connectivity from './Connectivity';
-import graphqlFragmentTypes from './generated/graphqlFragmentTypes.json';
 
-export default new ApolloClient({
+const httpLink = new HttpLink({
   uri: `${Config.api.origin}/--/graphql`,
+});
 
-  async request(operation): Promise<void> {
+const connectivityLink = setContext(
+  async (): Promise<any> => {
     const isConnected = await Connectivity.isAvailableAsync();
     if (!isConnected) {
       throw new Error('No connection available');
     }
+  }
+);
 
-    const { sessionSecret } = Store.getState().session;
-    if (sessionSecret) {
-      operation.setContext({
-        headers: { 'expo-session': sessionSecret },
-      });
-    }
+const authMiddlewareLink = setContext((): any => {
+  const { sessionSecret } = Store.getState().session;
+  if (sessionSecret) {
+    return {
+      headers: { 'expo-session': sessionSecret },
+    };
+  }
+});
+
+const link = connectivityLink.concat(authMiddlewareLink.concat(httpLink));
+
+const cache = new InMemoryCache({
+  possibleTypes: {
+    ActivityTimelineProjectActivity: ['Build', 'BuildJob'],
+    BuildOrBuildJob: ['Build', 'BuildJob'],
+    BaseSearchResult: ['UserSearchResult', 'AppSearchResult'],
+    Project: ['App', 'Snack'],
```

Comment:
What happens if we don't add a type here? It seems easy to forget to modify this.

Reply:
I think forgetting to add a type here would be the equivalent of forgetting to run the previous `generate-graphql-fragment-types.sh` when a new union type is added and used in the app. So while this doesn't improve the "forgetting to modify this", it also doesn't regress it in my opinion, and also makes things a bit simpler.

Reply:
Concretely I'm not sure though since the only union we use here is `Project` and even that one I don't think we use any more in this app.